### ISSUE TITLE
Provider produced inconsistent result after apply for ec_deployment and ec_deployment_traffic_filter_association

### ISSUE THREAD

**Issue by grumpymatt** [elastic/terraform-provider-ec#656]
## Readiness Checklist

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I am reporting the issue to the correct repository (for multi-repository projects)

## Expected Behavior

Terraform apply should return success.

## Current Behavior

Error after applying. Resources get created but result is an error.
```
| Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to ec_deployment.tms_elasticsearch[0], provider
│ "provider[\"registry.terraform.io/elastic/ec\"]" produced an unexpected new
│ value: .traffic_filter: was null, but now
│ cty.SetVal([]cty.Value{cty.StringVal("b0f348b752c94452bcd4472e70fd2b1b")}).
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
`````

## Terraform definition

````
data "ec_stack" "latest" {
  version_regex = "latest"
  region        = var.region
}

resource "ec_deployment" "deploy_elasticsearch" {
  count = var.qualifier == "qa" ? 0 : 1
  name = "deploy_${var.level_env}"

  region                 = var.region
  version                = data.ec_stack.latest.version
  deployment_template_id = "aws-general-purpose-arm-v6"

  elasticsearch = {
    hot = {
      autoscaling = {
      }
     size = "2g"
    }
  }

  kibana = {}

  enterprise_search = {}

  integrations_server = {}
}

data "ec_traffic_filter" "vpce_tf" {
  name = "${var.qualified_prefix}-elasticsearch-tf"
}

resource "ec_deployment_traffic_filter_association" "elasticsearch_traffic_filter_association" {
  count = var.qualifier == "qa" ? 0 : 1
  traffic_filter_id = data.ec_traffic_filter.vpce_tf.rulesets[0].id
  deployment_id     = ec_deployment.deploy_elasticsearch[0].id
}
````

## Steps to Reproduce

1. terraform apply

## Your Environment

* Version used: 
Terraform v1.4.6
on linux_amd64
provider registry.terraform.io/elastic/ec v0.7.0
provider registry.terraform.io/hashicorp/aws v4.62.0
provider registry.terraform.io/hashicorp/external v2.3.1
* Running against Elastic Cloud SaaS or Elastic Cloud Enterprise and version: Elastic Cloud
* Environment name and version (e.g. Go 1.9):
* Server type and version:
* Operating System and version: Linux
* Link to your project:

**Comment by tommed** [comment]
I'm getting this too. :-/
Happy to provide some details if that will help.
Obviously it's causing us some issues for production environments.

**Comment by renzedj** [comment]
Same error, and I'm not attempting to create a traffic filter.

**Comment by grumpymatt** [comment]
I found a potential workaround. Ignoring changes to the traffic_filter attribute on the ec_deployment allowed me to apply without errors.

Here is the resource with the ignore block at the bottom:
```# Create an Elastic Cloud deployment
resource "ec_deployment" "tms_elasticsearch" {
  count = var.qualifier == "qa" ? 0 : 1
  # Optional name.
  name = "ec_deploymentName"

  region                 = var.region
  version                = data.ec_stack.latest.version
  deployment_template_id = "aws-general-purpose-arm-v6"

  elasticsearch = {
    hot = {
      autoscaling = {
      }
     size = "2g"
    }
  }

  kibana = {}

  enterprise_search = {}

  integrations_server = {}

  lifecycle {
    ignore_changes = [
      traffic_filter
    ]
  }
}

**Comment by tommed** [comment]
> Ignoring changes to the traffic_filter attribute on the ec_deployment allowed me to apply without errors.

This isn't a good workaround for us as we want to track and apply the traffic filters.

Is there anyone from the provider developer community active on these boards? Or is plugin unsupported?

**Comment by grumpymatt** [comment]
Just so it's clear, I am still applying a traffic filter with my workaround. I am using the separate ec_deployment_traffic_filter_association resource to apply the traffic_filter and not the attribute on the ec_deployment.

**Comment by tommed** [comment]
Ah ok, thanks for the clarification - this is all quite new to me.
I did try it however, the workaround still results in the same message for me.

**Comment by tobio** [comment]
This appears to be a duplicate of https://github.com/elastic/terraform-provider-ec/issues/621. There's a fix for this in the main branch, which will be included in the next release. There's no timeframe on that at the moment, IIUC we're waiting on some ESS API changes in order to fix https://github.com/elastic/terraform-provider-ec/issues/433.
